talk/tools/profiling.tex

```diff
@@ -4,39 +4,228 @@
+      \item Powerful command line profiling tool for Linux
+      \item Not portable, the source code is part of the Linux kernel itself
+      \item Much lower overhead compared with \mintinline{bash}{valgrind}
+      \item In order to profile your code, make sure to compile with
+            \texttt{CXXFLAGS="-O2 -g -fno-omit-frame-pointer"}
+      \item Counting and sampling
+        \begin{itemize}
+          \item Counting -- count occurrences of a given event (e.g.\ cache misses)
+          \item Time-based sampling -- sample the stack at regular time intervals
+          \item Event-based sampling -- take samples when event counter overflows
+          \item Instruction-based sampling -- sample instructions and precisely count events they create
+        \end{itemize}
+      \item Static and dynamic tracing
+        \begin{itemize}
+          \item Static -- pre-defined tracepoints in software (e.g.\ scheduling events)
+          \item Dynamic -- tracepoints created dynamically with \mintinline{bash}{perf probe}
+        \end{itemize}
     \end{itemize}
-  \end{block}
-  \begin{block}{Intel VTune and AMD uProf}
-    \begin{itemize}
-      \item Graphical profilers from CPU vendors with rich features
-      \item Needs vendor's CPU for full experience
-      \item More information on \href{https://www.intel.com/content/www/us/en/developer/tools/oneapi/vtune-profiler.html}{Intel's website} and \href{https://developer.amd.com/amd-uprof/}{AMD's website}
-    \end{itemize}
-  \end{block}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitle{\mintinline{bash}{perf} commands}
+  { \scriptsize
+    \begin{block}{}
+      \begin{minted}{shell-session}
+$ perf
+ usage: perf [--version] [--help] [OPTIONS] COMMAND [ARGS]
+ The most commonly used perf commands are:
+   annotate        Read perf.data and display annotated code
+   c2c             Shared Data C2C/HITM Analyzer.
+   config          Get and set variables in a configuration file.
+   diff            Read perf.data and display the differential profile
+   evlist          List the event names in a perf.data file
+   list            List all symbolic event types
+   mem             Profile memory accesses
+   record          Run a command and record its profile into perf.data
+   report          Read perf.data and display the profile
+   sched           Tool to trace/measure scheduler properties (latencies)
+   script          Read perf.data and display trace output
+   stat            Run command and gather performance counter statistics
+   top             System profiling tool.
+   version         display the version of perf binary
+   probe           Define new dynamic tracepoints
+   trace           strace inspired tool
+ See 'perf help COMMAND' for more information on a specific command.
+      \end{minted}
+    \end{block}
+  }
+\end{frame}
```

Comment:
Is this useful ? I think I would drop it

Reply:
I use a similar slide to this to give a general overview of perf, mentioning that there are more commands than the ones I cover, but it's probably fine to drop. I did have to shorted the description of the commands to fit in the slide anyway, so this is not quite what you'd get by running perf without arguments.

Reply:
On first thought I also found this too much. On second thought, yeah, why shouldn't we leave an overview here.

Reply:
The point is that this slide would be systematically skipped when you present. So if it's a pure reference, then let's put it in a reference section at the very end. Otherwise, let's drop it.
> mentioning that there are more commands than the ones I cover

Useful indeed, but then I would mention that there are a lot of commands, not list them

Reply:
It seems that most people don't think it's useful, so I will drop this slide.